src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/applicationScopes.jsx

```diff
@@ -0,0 +1,101 @@
+import {Flex} from 'grid-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+import Switch from 'app/components/switch';
+
+import {API_SCOPES} from 'app/constants';
+
+class ApplicationScopes extends React.Component {
+  static propTypes = {
+    scopes: PropTypes.arrayOf(PropTypes.string),
+    onToggle: PropTypes.func,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      error: false,
+      scopes: new Set(this.props.scopes),
+    };
+  }
+
+  handleToggleScopes = (scope, e) => {
+    let {scopes} = this.state;
+    if (scopes.has(scope)) {
+      scopes.delete(scope);
+    } else {
+      scopes.add(scope);
+    }
+
+    this.setState(
+      state => {
```

Comment:
The above logic (L25-30) needs to be in here. The idea is that because `setState` is async, state could have changed by the time it is ready to update state.